### DISCUSSION TITLE
Mitigating Data Loss Risks in Schema while Applying Changes in a Dual Database Environment

### DISCUSSION THREAD

**Discussion by bamidele2161** [prisma/prisma#22915] [Q&A]
### Question

In a scenario where two different databases are being used, each with its schema (Schema A and Schema B), and there's a need to make changes to fields of a table in Schema B, the challenge arises when running the migration command:

npx prisma migrate dev --schema prisma/schema2.prisma --name edit-form-datable --create-only

Even though the --create-only flag is used to generate migration files without applying the changes to the database immediately, it still prompts with a warning: "Do you want to continue? All data will be lost."

In this case, what steps can be taken to prevent potential data loss when making changes to Schema B?

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @bamidele2161 👋 

Thank you for raising this question.

When using the `--create-only` flag,  it's important to note that while the new migration won't be applied, previous migrations will be applied, which can lead to potential data loss. Can you check the[ `_prisma_migrations`](https://www.prisma.io/docs/orm/prisma-migrate/understanding-prisma-migrate/migration-histories#migration-history) table in your database to see if any of your migrations is still in `pending` state?

If a migration is in pending state, you will need to manually [resolve](https://www.prisma.io/docs/orm/prisma-migrate/workflows/patching-and-hotfixing#failed-migration) it.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
